Bottom: a9c8bda415a685c39f31a6cb94fc8dd19a9c0325
Top:    7ba15fc6ca7058a9b4b0e083d2e7d9e131417753
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-09-05 16:11:37 +0100

hid/gtk: Use gtk_notebook_get_n_pages() rather than diving in and counting

In GTK3.0, direct access to notebook->children will be impossible.


---

diff --git a/src/hid/gtk/gui-config.c b/src/hid/gtk/gui-config.c
index 851571a..698f57a 100644
--- a/src/hid/gtk/gui-config.c
+++ b/src/hid/gtk/gui-config.c
@@ -1980,7 +1980,7 @@ config_page_create (GtkTreeStore * tree, GtkTreeIter * iter,
 
   vbox = gtk_vbox_new (FALSE, 0);
   gtk_notebook_append_page (notebook, vbox, NULL);
-  page = g_list_length (notebook->children) - 1;
+  page = gtk_notebook_get_n_pages (notebook) - 1;
   gtk_tree_store_set (tree, iter, CONFIG_PAGE_COLUMN, page, -1);
   return vbox;
 }
